### PR TITLE
double higgs analyser 

### PR DESCRIPTION
**Pull request by YooSunYoung**
instructions are in analysis/test/hh/recipe....sh file

### CODE REVIEW

File: analysis/bin/doubleHiggsAnalyser.cc [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+bool doubleHiggsAnalyser::NanoAnalysis() {
+  return false;
+}
+
+bool doubleHiggsAnalyser::DelphesAnalysis() {
+  //map<Float_t, int, greater<Float_t>> muons : map of <pt,index>:<K,V> of muon sorted by pt.
+    
+    auto m = static_cast<const MissingET *>(missings->At(0)); // There is always one MET object.
+    if (m->MET<20) return false;
+    missing.SetPtEtaPhiM(m->MET,0,m->Phi,0.19);
```
Comment (jshlee) [inline analysis/bin/doubleHiggsAnalyser.cc]: why is mass set to 0.19?

File: analysis/bin/doubleHiggsAnalyser.cc [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    if (from2==25) fromHiggs++;
+    if (from2==6) fromTop++;
+    if (from2==32) fromZ++;
+    muon2.SetPtEtaPhiM(mu2->PT,mu2->Eta,mu2->Phi,mu2->Mass);
+    
+    lepton1_pt = mu1->PT;
+    lepton2_pt = mu2->PT;
+    missing_et = m->MET;
+    auto muons = muon1+muon2;
+    mt = muons.M()*muons.M() + muons.Px()*muons.Px() + muons.Py()*muons.Py();
```
Comment (jshlee) [inline analysis/bin/doubleHiggsAnalyser.cc]: cant you just do muons.Mt()

File: analysis/bin/doubleHiggsAnalyser.cc [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    if (muons.size()<2) {
+      return false;
+    }
+    
+    map<Float_t,int,greater<Float_t>>::iterator iter = muons.begin();
+    auto mu1 = static_cast<const GenParticle *>(particles->At(iter->second));
+    int from1 = isFrom(particles, iter->second);
+    from1 = abs(from1);
+    if (from1==25) fromHiggs++;
+    if (from1==6) fromTop++;
```
Comment (jshlee) [inline analysis/bin/doubleHiggsAnalyser.cc]: shouldnt this be abs?

File: analysis/bin/doubleHiggsAnalyser.cc [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+    if (from1==32) fromZ++;
```
Comment (jshlee) [inline analysis/bin/doubleHiggsAnalyser.cc]: Z is 23 no?

File: analysis/bin/doubleHiggsAnalyser.cc [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+    
+    lepton1_pt = mu1->PT;
+    lepton2_pt = mu2->PT;
+    missing_et = m->MET;
+    auto muons = muon1+muon2;
+    if (muons.M() > 65 || muon1.DeltaR(muon2) > 1) {
+        return false;
+    }
+    mt = muons.M()*muons.M() + muons.Px()*muons.Px() + muons.Py()*muons.Py();
+
```
Comment (jshlee) [inline analysis/bin/doubleHiggsAnalyser.cc]: isnt this the same as mouns.Mt()?

File: analysis/bin/doubleHiggsAnalyser.cc (hunk 3 above)
Comment (YooSunYoung) [reply analysis/bin/doubleHiggsAnalyser.cc]: doesn't from1 = abs(from1) work....?

File: analysis/bin/doubleHiggsAnalyser.cc (hunk 4 above)
Comment (YooSunYoung) [reply analysis/bin/doubleHiggsAnalyser.cc]: Sorry.. I was confused Z' with Z0 I'll make those id as static constant variables

File: analysis/bin/doubleHiggsAnalyser.cc [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+    map<Float_t,int,greater<Float_t>>::iterator iter = muons.begin();
+    auto mu1 = static_cast<const GenParticle *>(particles->At(iter->second));
+    int from1 = isFrom(particles, iter->second);
+    from1 = abs(from1);
+    if (from1==25) fromHiggs++;
+    if (from1==6) fromTop++;
+    if (from1==32) fromZ++;
+    muon1.SetPtEtaPhiM(mu1->PT,mu1->Eta,mu1->Phi,mu1->Mass);
+    ++iter;
+    auto mu2 = static_cast<const GenParticle *>(particles->At(iter->second));
```
Comment (jshlee) [inline analysis/bin/doubleHiggsAnalyser.cc]: is static cast needed?

File: analysis/bin/doubleHiggsAnalyser.cc (hunk 5 above)
Comment (YooSunYoung) [reply analysis/bin/doubleHiggsAnalyser.cc]: I thought so ... but they weren't same so I was trying to see the distribution for each of them.

File: analysis/bin/doubleHiggsAnalyser.cc (hunk 6 above)
Comment (YooSunYoung) [reply analysis/bin/doubleHiggsAnalyser.cc]: then should I change static_cast to dynamic_cast...?

Comment (jshlee) [reply analysis/bin/doubleHiggsAnalyser.cc]: why cast in the first place?

File: analysis/bin/doubleHiggsAnalyser.cc (hunk 5 above)
Comment (jshlee) [reply analysis/bin/doubleHiggsAnalyser.cc]: you should be doing the sqrt of this

File: analysis/bin/doubleHiggsAnalyser.cc (hunk 6 above)
Comment (YooSunYoung) [reply analysis/bin/doubleHiggsAnalyser.cc]: cause something like ```mu2->PT;``` doesn't work without casting to GenParticle type ...?

**Comment by jshlee** [comment]
+1

**Pull request merged**